pkgs/by-name/do/dosbox-staging/package.nix

```diff
@@ -136,7 +88,10 @@ stdenv.mkDerivation (finalAttrs: {
       comment = "x86 dos emulator enhanced";
       desktopName = "DosBox-Staging";
       genericName = "DOS emulator";
-      categories = [ "Emulator" "Game" ];
+      categories = [
+        "Emulator"
+        "Game"
+      ];
     })
   ];
```

Comment:
Can we replace this with the [upstream desktop file](https://github.com/dosbox-staging/dosbox-staging/blob/main/contrib/linux/dosbox-staging.desktop)?

Reply:
Good idea. Done :)

Reply:
We probably actually want to name it dosbox-staging.desktop, do a substituteInPlace of the desktop file to set the exec to be `dosbox-staging` and set meta.mainProgram to `dosbox-staging` so there are no conflicts in case users want all the different DOSBoxes available in nixpkgs.

Also, looks like the build is failing now:

```
error: builder for '/nix/store/94nln40gaff7smqn12pri1zz033di0gi-dosbox-staging-0.81.0.drv' failed with exit code 1;
       last 25 log lines:
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_3848.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_20866.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28591.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28592.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28593.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28594.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28595.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28596.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28597.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28598.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28600.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28601.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28603.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28604.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28606.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/translations/de.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/en.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/es.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/fr.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/it.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/nl.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/pl.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/ru.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > cp: cannot stat 'contrib/linux/dosbox-staging.desktop': No such file or directory
       > /nix/store/558iw5j1bk7z6wrg8cp96q2rx03jqj1v-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/94nln40gaff7smqn12pri1zz033di0gi-dosbox-staging-0.81.0.drv'.
error: 1 dependencies of derivation '/nix/store/f049fqaxblcqbax04rfqlpl0p4677pk9-review-shell.drv' failed to build
```

Reply:
Fixed. Thanks.